test/metabase/lib/filter_test.cljc

```diff
@@ -181,3 +191,80 @@
                                 [:= venues-category-id-metadata 242 categories-id-metadata]
                                 [:contains venues-name-metadata "part"]]])
                   (dissoc :lib/metadata)))))))
+
+(deftest ^:parallel add-filter-test
+  (let [q1                   (lib/query-for-table-name meta/metadata-provider "CATEGORIES")
+        venues-name-metadata (lib.metadata/field q1 nil "VENUES" "NAME")]
+    (let [simple-query
+          {:lib/type :mbql/query,
+           :database (meta/id)
+           :type :pipeline
+           :stages [{:lib/type :mbql.stage/mbql
+                     :source-table (meta/id :categories)
+                     :lib/options {:lib/uuid (str (random-uuid))}}]}
+          first-filter
+          [:between
+           {:lib/uuid string?}
+           [:field
+            {:base-type :type/Integer, :lib/uuid string?}
+            (meta/id :venues :category-id)]
+           42
+           100]
```

Comment:
It would be good to dogfood stuff in `metabase.lib.core` a bit and try to use those to build queries instead of doing it by hand